src/pages/index.js

```diff
@@ -5,123 +5,260 @@ import PopupWithForm from "../components/PopupWithForm.js";
-  const { name, description } = userInfo.getUserInfo();
-  nameInput.value = name;
-  jobInput.value = description;
-  editFormValidator.resetValidation();
+addNewCardButton.addEventListener("click", () => {
+  newCardPopup.open();
+});
+
+editAvatarButton.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
+confirmDeleteBtn.addEventListener("click", () => {
+  cardDeletePopup.close();
+});
+
+profileEditBtn.addEventListener("click", (e) => {
   editProfilePopup.open();
+  e.preventDefault();
+  const newUserInfo = userInfo.getUserInfo();
+  document.querySelector("#profile-name-input").value = newUserInfo.name;
+  document.querySelector("#profile-description-input").value =
+    newUserInfo.description;
+});
+
+// Functions
+// Functions
+
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteModal,
+    handleLikeCard
+  );
+  const cardElement = card.getView();
+  return cardElement;
 }
 
-function handleImageClick(cardData) {
-  newImagePopup.open(cardData);
+function handleDeleteModal(card) {
+  cardDeletePopup.submitHandler(() => handleDeleteCard(card));
+  cardDeletePopup.open();
 }
 
-function handleAddCardFormSubmit(userInfo) {
-  const name = userInfo.title;
-  const link = userInfo.url;
-  renderCard({ name, link }, cardListElement);
-  addCardFormElement.reset();
-  newCardPopup.close();
-  addFormValidator.disableButton();
+function handleDeleteCard(card) {
+  const cardId = card.gatherCardId();
+  api
+    .cardDelete(cardId)
+    .then(() => {
+      card.handleDelete();
+      //cardDeletePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
 }
 
-//Event Listeners
-//Event Listeners
+function handleLikeCard(card) {
+  if (card.isLiked) {
+    api
+      .dislikeCard(card.gatherCardId())
+      .then((res) => {
+        card.handleLike(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .likeCard(card.gatherCardId())
+      .then((res) => {
+        card.handleLike(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
 
-profileEditBtn.addEventListener("click", prefillProfileData);
+// Form Submission
+// Form Submission
 
-addNewCardButton.addEventListener("click", () => newCardPopup.open());
+function handleAddCardFormSubmit(data) {
+  newCardPopup.renderloading(true);
+  api
+    .newCardAdd({ name: data.title, link: data.url })
+    .then((data) => {
+      const addCardElement = createCard(data);
+      section.addItem(addCardElement);
+      newCardPopup.resetForm();
+      newCardPopup.close();
+      addFormValidator.disableButton();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      newCardPopup.renderloading(false);
+    });
+}
 
-//Event Handlers
-//Event Handlers
+function handleProfileEditSubmit(userData) {
+  editProfilePopup.renderloading(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```